src/toil/lib/misc.py

```diff
@@ -118,16 +119,21 @@ def AtomicFileCreate(final_path, keep=False):
                 pass
         raise
 
-def atomic_copy(src_path, dest_path):
+def atomic_copy(src_path, dest_path, executable=False):
     """Copy a file using posix atomic creations semantics."""
     with AtomicFileCreate(dest_path) as dest_path_tmp:
         shutil.copyfile(src_path, dest_path_tmp)
+    if executable:
+        os.chmod(dest_path, os.stat(dest_path).st_mode | stat.S_IXUSR)
 
-def atomic_copyobj(src_fh, dest_path, length=16384):
+def atomic_copyobj(src_fh, dest_path, length=16384, executable=False):
     """Copy an open file using posix atomic creations semantics."""
     with AtomicFileCreate(dest_path) as dest_path_tmp:
         with open(dest_path_tmp, 'wb') as dest_path_fh:
             shutil.copyfileobj(src_fh, dest_path_fh, length=length)
+    if executable:
```

Comment:
As with the function above, we should set the permissions inside of the `with AtomicFileCreate` block, operating on `dest_path_tmp`.